elements/lisk-p2p/src/p2p.ts

```diff
@@ -759,26 +764,40 @@ export class P2P extends EventEmitter {
 			random,
 			Math.min(minimumPeerDiscoveryThreshold, knownPeers.length),
 		);
-		/* tslint:enable no-magic-numbers*/
-
-		// TODO: Remove fields that are specific to the current Lisk protocol.
-		const peers = this._pickRandomPeers(randomPeerCount).map(
-			(peerInfo: P2PPeerInfo): ProtocolPeerInfo => {
-				const { ipAddress, ...peerInfoWithoutIp } = peerInfo;
 
-				// The options property is not read by the current legacy protocol but it should be added anyway for future compatibility.
-				return {
-					...peerInfoWithoutIp,
-					ip: ipAddress,
-				};
-			},
-		);
-		const protocolPeerInfoList: ProtocolPeerInfoList = {
-			success: true,
-			peers,
-		};
-
-		request.end(protocolPeerInfoList);
+		if (request.procedure === REMOTE_RPC_GET_MINIMAL_PEERS_LIST) {
```

Comment:
As mentioned above, we can use `REMOTE_RPC_GET_MINIMAL_PEERS_LIST ` for `getPeers` RPC for the 2.3 nodes and else part to support old versions